### PR TITLE
#4549 kotlin ijar test cases

### PR DESCRIPTION
**Pull request by hsyed**
Inlining test cases for #4549. The jar also contains the metadata file which needs to be retained `META-INF/inline-cases.kotlin_module`

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: third_party/ijar/test/ijar_test.sh [hunk 1]
```diff
@@ -114,13 +116,49 @@ function check_consistent_file_contents() {
...
 function tear_down() {
   rm -fr $TEST_TMPDIR/classes
+  rm -fr $TEST_TMPDIR/kotlin
+}
+
+function run_ijar_and_extract_to() {
+    mkdir -p $3 || fail "could not ensure output dir"
+    ${IJAR} $1 $2 || fail "ijar failed"
+    pushd $3
+        ${JAR} xf $2 || fail "could not extract jar contents"
```
Comment (philwo) [inline third_party/ijar/test/ijar_test.sh]: Please just indent with 4 spaces here, too.

File: third_party/ijar/test/kotlin/InlineCases.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+object ObjectWithInline {
+    inline fun objectInlineFun(op: () -> Unit) { op() }
+}
+
+abstract class AbstractClassWithInline {
+    inline fun inheritedInlineFun(op: () -> Unit) { op() }
+}
+
+object ObjectInheritingInline: AbstractClassWithInline()
```
Comment (philwo) [inline third_party/ijar/test/kotlin/InlineCases.kt]: Please add newline at the end of the file.

File: third_party/ijar/test/kotlin/regen.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash -eu
+
+kotlinc -module-name inline-cases -d inline-cases.jar InlineCases.kt
```
Comment (philwo) [inline third_party/ijar/test/kotlin/regen.sh]: Please add newline at the end of the file.

File: third_party/ijar/test/ijar_test.sh (hunk 1 above)
Comment (hsyed) [reply third_party/ijar/test/ijar_test.sh]: It's showing up as 4 spaces on my side (not a tab).

**Comment by laszlocsomor** [comment]
@hsyed : Do you have any updates? Do you still want to merge this PR?

**Comment by laszlocsomor** [comment]
@philwo , could you please take another look?

**Comment by laszlocsomor** [comment]
Thanks @philwo for the update! Do you want to import this?

**Comment by philwo** [comment]
@laszlocsomor Yes, I'll send you a CL. :)

**Comment by hsyed** [comment]
It is ready to merge

**Comment by buchgr** [comment]
@hsyed Sorry for the delay. Philipp needs to make some changes internally so that he can import this PR.

**Comment by philwo** [comment]
@hsyed I tried to import this PR, but the tests don't pass. I see the following errors. Can you help me figuring out what's wrong?

You can see the current version of the code that I tried to check in here: https://bazel-review.googlesource.com/c/bazel/+/44791 (please ignore any diffs in files except third_party/ijar/... - they are unfortunately generated by our importing script and are just a UI issue)


```
** test_kotlin_metadata_retention **********************************************
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/436426416158653421/execroot/io_bazel/_tmp/6509e9299f97f59c74a8df033517ef10/kotlin/inline /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/436426416158653421/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/436426416158653421/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel
-- Test log: -----------------------------------------------------------
------------------------------------------------------------------------
test_kotlin_metadata_retention FAILED: kotlin metadata not retained .
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/436426416158653421/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel/third_party/ijar/test/ijar_test:143: in call to test_kotlin_metadata_retention
FAILED: test_kotlin_metadata_retention
```

see https://storage.googleapis.com/bazel-buildkite-artifacts/f705022d-d8a8-41bc-8910-61ed9770b060/third_party/ijar/test/ijar_test/attempt_1.log

```
** test_kotlin_inline_retention ************************************************
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/7198431800070087349/execroot/io_bazel/_tmp/80b3b032c13a55d5f62efcda1a9bedd9/kotlin/inline /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/7198431800070087349/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/7198431800070087349/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel
-- Test log: -----------------------------------------------------------
public final class ClassWithInline {
  public final void classInlineFun(kotlin.jvm.functions.Function0<kotlin.Unit>);

  public ClassWithInline();
}
------------------------------------------------------------------------
test_kotlin_inline_retention FAILED: javap output missing expected substring:   public final void classInlineFun(kotlin.jvm.functions.Function0<kotlin.Unit>);
    Code: .
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/7198431800070087349/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel/third_party/ijar/test/ijar_test:137: in call to javap_decompile_and_assert_contains
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/d4fe08e9d117c0a3a25bf2b3039e8671/bazel-sandbox/7198431800070087349/execroot/io_bazel/bazel-out/k8-fastbuild/bin/third_party/ijar/test/ijar_test.runfiles/io_bazel/third_party/ijar/test/ijar_test:148: in call to test_kotlin_inline_retention
FAILED: test_kotlin_inline_retention
```

see https://storage.googleapis.com/bazel-buildkite-artifacts/f705022d-d8a8-41bc-8910-61ed9770b060/third_party/ijar/test/ijar_test/attempt_4.log

**Comment by philwo** [comment]
@hsyed Friendly ping - I would love to get this merged.

**Comment by hsyed** [comment]
@philwo sorry for the day. The test cases are for #4549 they will fail till the ijar tool is updated.

**Comment by irengrig** [comment]
Hi All, Bazel sheriff is here. What is the status of this?

**Comment by hsyed** [comment]
This isn't in scope any longer. I'll close it.

**Pull request closed without merging**